covalent/_file_transfer/strategies/s3_strategy.py

```diff
@@ -0,0 +1,84 @@
+import os
+import urllib.request
+from pathlib import Path
+from subprocess import PIPE, CalledProcessError, Popen
+from typing import Optional
+
+from furl import furl
+
+from covalent._file_transfer import File
+from covalent._file_transfer.strategies.transfer_strategy_base import FileTransferStrategy
+
+
+class S3(FileTransferStrategy):
+
+    """
+    Implements Base FileTransferStrategy class to upload/download files from S3 Bucket.
+    """
+
+    def __init__(
+        self,
+            credentials: str = os.environ.get("AWS_SHARED_CREDENTIALS_FILE") or os.path.join(os.environ["HOME"], ".aws/credentials"),
+            profile: str = os.environ.get("AWS_PROFILE") or None,
+            region_name: str = os.getenv("AWS_REGION") or None,
+    ):
+
+        self.credentials = credentials
+        self.profile = profile
+        self.region_name = region_name
+
+        try:
+            import boto3
+        except ImportError:
+            raise ImportError("Using S3 strategy requires boto3 from AWS installed on your system.")
```

Comment:
@poojithurao seems some tests fail when instantiating the S3 class as `boto3` is not installed. could you update tests to mock out boto3 (as we do not actually invoke its methods) ? 

refer to: https://stackoverflow.com/questions/32508349/python-mock-and-libraries-that-are-not-installed

Reply:
@AlejandroEsquivel I have updated the tests now.